Avoid allocation for ChainNodes#_paths

stefanpenner opened the pull request:
* mostly this structure is empty (checked afew sites only 5%-7% of chain nodes have members in this property)
* don’t allocate it eagerly
* short-circuit accessing it if it is not present (except for additions of course)

stefanpenner committed:
[BUGFIX beta] Avoid allocation for ChainNodes#_paths

* mostly this structure is empty
* don’t allocate it eagerly
* short-circuit accessing it if it is not present (except for additions of course)

krisselden commented:
Good catch, paths is only a root node thing. Should this target beta? This area has low churn, easy to backport

stefanpenner commented:
updated to target beta

krisselden commented:
awesome!

stefanpenner commented:
```
- Original
- Updated (with the patch)
running first test, please wait...
  Original ... 14,315,166.10 op/s
  Updated .... 19,221,458.12 op/s
```
